tests/testflows/rbac/tests/syntax/alter_settings_profile.py

```diff
@@ -0,0 +1,232 @@
+from contextlib import contextmanager
+
+from testflows.core import *
+
+from rbac.requirements import *
+import rbac.tests.errors as errors
+
+@TestFeature
+@Name("alter settings profile")
+@Args(format_description=False)
+def feature(self, node="clickhouse1"):
+    """Check alter settings profile query syntax.
+
+    ```sql
+    ALTER SETTINGS PROFILE [IF EXISTS] name
+    [ON CLUSTER cluster_name]
+    [RENAME TO new_name]
+    [SETTINGS variable [= value] [MIN [=] min_value] [MAX [=] max_value] [READONLY|WRITABLE] | INHERIT 'profile_name'] [,...]
+    [TO {user_or_role [,...] | NONE | ALL | ALL EXCEPT user_or_role [,...]]}
+    ```
+    """
+    node = self.context.cluster.node(node)
+
+    def cleanup_profile(profile):
+        with Given(f"I ensure that profile {profile} does not exist"):
+            node.query(f"DROP SETTINGS PROFILE IF EXISTS {profile}")
+
+    try:
+        with Given("I have a profile and some users and roles"):
+            node.query(f"CREATE SETTINGS PROFILE profile0")
+            node.query(f"CREATE USER user0")
+            node.query(f"CREATE ROLE role0")
+
+        with Scenario("I alter settings profile with no options", flags=TE, requirements=[RQ_SRS_006_RBAC_SettingsProfile_Alter("1.0")]):
+            with When("I alter settings profile"):
+                node.query("ALTER SETTINGS PROFILE profile0")
+        
+        with Scenario("I alter settings profile short form", flags=TE, requirements=[RQ_SRS_006_RBAC_SettingsProfile_Alter("1.0")]):
+            with When("I short form alter settings profile"):
+                node.query("ALTER PROFILE profile0")
+
+        with Scenario("I alter settings profile that does not exist, throws exception", flags=TE, requirements=[RQ_SRS_006_RBAC_SettingsProfile_Alter("1.0")]):
```

Comment:
What will happen if we edit the requirements.md and remove some requirements there which are still referenced in our tests in *.py files?

Reply:
Unless you regenerate the `requirements.py` using the updated `requirements.md` then nothing will happen. If we do regenerate and get a new `requirements.py` that does not define requirements that are referenced in tests then you will get an `AttributeError` or something like it when you try to run the tests.